### PR TITLE
Component/select

### LINKED ISSUE #10: Select
Built a Select component.

Screenshot 2021-01-09 at 10 20 55

View the component on [Figma](https://www.figma.com/file/gGwSe7ilZCUmAIaJyUc4iY/Bristol-Tech-Fair-Design-System?node-id=489%3A44).

### PR DESCRIPTION
**Pull request by vixhez**
Closes #10 

Made some progress with the select component. Initial styling - border, basic layout, padding inside component. Work still to be done on drop down arrow styling.

### CODE REVIEW

**Comment by RobTF9** [comment]
Good progress! Before I recommend changes though can you target this pull request at the `develop` branch as opposed to `main`?

**Comment by vixhez** [comment]
All sorted!

File: src/components/Select/Select.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      </SelectDropdown>
+    </SelectWrapper>
+  );
+};
+
+Select.propTypes = {
+  value: PropTypes.string.isRequired
+};
+
+export default Select;
```
Comment (RobTF9) [inline src/components/Select/Select.jsx:24]: A few issues with this file: 
• Don't hard code the options in like that, pass them in as an array and map over them.
• I would change `SelectDropdown` to a `select` element and style it with a nested selector (see comments on Selectstyles.js).
• The `value` prop is not a value, rather the label text. Consider changing this `label`.
• Don't hard code the name of the select, pass it in as a prop called `name`.
• You need to pass in an `onChange` handler.
• Pass in a `defaultValue` prop.

So with the above in mind it should end up looking something like this:
```
const Select = ({ label, name, options, onChange, defaultValue }) => {
    return (
        <SelectWrapper>
            <label htmlFor={name}>{label}</label>
            <select
                name={name}
                id={name}
                onChange={onChange}
                value={defaultValue}
            >
                {options.map(({ value, label }, index)  =>  (
                     <option
                        key={value + index}
                        className="select--option"
                        value={value}
                       >
                          {label}
                     </option>
                ))}
            </select>
        </SelectWrapper>
    )
};

Select.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  defaultValue: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  onChange: PropTypes.func.isRequired,
};
```

File: src/components/Select/Select.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import Select from './Select';
+
+export default {
+  title: 'Components/Select',
+  component: Select
+};
+
+export const HelloWorld = () => <Select value="Example label" />;
```
Comment (RobTF9) [inline src/components/Select/Select.stories.jsx:9]: Update this file to show a few variations using the different props you add in the main jsx file.

File: src/components/Select/Select.styles.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  display: block;
+  border: 0.1rem solid var(--lightblue);
+  border-radius: 4px;
+  background-color: var(--lightgrey);
+  color: var(--darkgrey);
+  padding: var(--smallest-space);
+  padding-left: 0;
+  width: 100%;
+  max-width: 500px;
+`;
```
Comment (RobTF9) [inline src/components/Select/Select.styles.js:18]: As mention in comments above a few changes to make here:

- You shouldn't use pixel values, we're using rem for sizing so change border-radius to `0.4rem`.
- Don't need a max-width, width, padding left or display.
- Nest the select in `SelectWrapper`.
- Create a dropdown arrow with the `::after` pseudo select (https://css-tricks.com/snippets/css/css-triangle/ great resource for creating triangles with css.
- Remove default dropdown arrow with `-moz-appearance: none;` and `-webkit-appearance: none;` 
- Position the items within the `SelectWrapper` with flexbox.
- Add correct padding values to select.

You should end up with something like this:
```
import styled from 'styled-components';

export const SelectWrapper = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;

  select {
    font-size: 1.6rem;
    border: 0.1rem solid var(--lightblue);
    border-radius: 0.4rem;
    background-color: var(--lightgrey);
    color: var(--darkgrey);
    padding: var(--small-space) var(--smallest-space);
    -moz-appearance: none;
    -webkit-appearance: none;
  }

  &::after {
    content: ' ';
    display: block;
    position: absolute;
    top: 5.2rem;
    right: var(--smallest-space);
    width: 0;
    height: 0;
    border-left: 0.8rem solid transparent;
    border-right: 0.8rem solid transparent;
    border-top: 1.2rem solid var(--lightblue);
  }
`;
```
If you would like me to go through in more detail just let me know.

**Review by RobTF9 — CHANGES_REQUESTED** [review]

**Comment by vixhez** [comment]
Thank you Rob! I will look into all of these ahead of Wednesday.

**Pull request closed without merging**